packages/generator/src/gulpfileHelpers.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from "path";
+import _ from "lodash";
+
+import { diffRaml, NodeDiff } from "@commerce-apps/raml-toolkit";
+import { generatorLogger } from "./logger";
+
+export type RamlDiff = {
+  file: string;
+  message?: string;
+  diff?: NodeDiff[];
+};
+
+/**
+ * Extracts all the RAML file names from the specified config file.
+ *
+ * @param configPath - Target config file
+ *
+ * @returns An array of `assetId/fileName` for every api in the input file
+ */
+export function listRamlsFromConfig(configPath: string): string[] {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const config = require(configPath);
+  const ramls = [];
+  for (const apiFamily of Object.keys(config)) {
+    for (const api of config[apiFamily]) {
+      ramls.push(path.join(api.assetId, api.fatRaml.mainFile));
+    }
+  }
+
+  return ramls;
+}
+
+/**
+ * Finds differences between the given directories for all the raml files in the
+ * provided config.
+ *
+ * @param oldApiDir - Existing APIs
+ * @param newApiDir - Newly downloaded APIs
+ * @param oldConfigPath - Existing config
+ * @param newConfigPath - Config for newly created APIs
+ *
+ * @returns An array of RamlDiff objects containing differences between all the
+ * old and new RAML files
+ */
+export async function diffNewAndArchivedRamlFiles(
+  oldApiDir: string,
+  newApiDir: string,
+  oldConfigPath: string,
+  newConfigPath: string
+): Promise<RamlDiff[]> {
+  const oldRamls = listRamlsFromConfig(oldConfigPath);
+  const newRamls = listRamlsFromConfig(newConfigPath);
+
+  const commonRamls = [];
+  // Extract all RAMLs that are part of the existing and the newly downloaded
```

Comment:
Probably don't want to extract as part of this method so it is independently testable.